WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java

```diff
@@ -472,9 +472,12 @@ else if (mPost.getDate_created_gmt() > 0)
             mPost.setLocation(mPostLocation);
         }
 
+        if (mCategories != null) {
+            mPost.setJSONCategories(new JSONArray(mCategories));
+        }
```

Comment:
I'm trying to figure out how it got to be null in the first place as it's never explicitly set to null. If it's just in a bad state I don't think we'd want to delete the existing categories. All I've got to go off is that `updatePostSettings` was called from the auto-save background timer in `EditPostActivity`.